fortran/spglib_f08.F90

```diff
@@ -725,6 +735,59 @@ end subroutine spg_free_dataset_c
 
    end function spg_get_dataset
 
+   function spg_get_magnetic_spacegroup_type(uni_number) result(magspgtype)
+    integer(c_int), intent(in) :: uni_number
+    type(SpglibMagneticSpacegroupType) :: magspgtype
+
+    type, bind(c) :: SpglibMagneticSpacegroupType_c
+       integer(c_int) :: uni_number
+       integer(c_int) :: litvin_number
+       character(kind=c_char) :: bns_number(8)
+       character(kind=c_char) :: og_number(12)
+       integer(c_int) :: number
+       integer(c_int) :: type
+    end type SpglibMagneticSpacegroupType_c
+
+    interface
+
+       function spg_get_magnetic_spacegroup_type_c(uni_number_c) &
+            & bind(c, name='spg_get_magnetic_spacegroup_type')
+          import c_int, SpglibMagneticSpacegroupType_c
+          integer(c_int), intent(in), value :: uni_number_c
+          type(SpglibMagneticSpacegroupType_c) :: spg_get_magnetic_spacegroup_type_c
+       end function spg_get_magnetic_spacegroup_type_c
+
+    end interface
+
+    type(SpglibMagneticSpacegroupType_c):: magspgtype_c
+    integer :: i
+
+    magspgtype_c = spg_get_magnetic_spacegroup_type_c(uni_number)
+
+    magspgtype%uni_number = magspgtype_c%uni_number
+    magspgtype%litvin_number = magspgtype_c%litvin_number
+
+    do i = 1, size(magspgtype_c%og_number)
+        if (magspgtype_c%og_number(i) == C_NULL_CHAR) then
+           magspgtype%og_number(i:) = ' '
+           exit
+        end if
+        magspgtype%og_number(i:i) = magspgtype_c%og_number(i)
+     end do
```

Comment:
Does `trim` not work with `C_NULL_CHAR`?

Reply:
I have no idea how to do it, maybe due to character types are different on left and right hand sides.

Reply:
I.e.:
```fortran
magspgtype%og_number = trim(magstype_c%og_number)
```
Does this work?

Reply:
I didn't work for me.
```
[ 50%] Building Fortran object fortran/CMakeFiles/Spglib_fortran.dir/spglib_f08.F90.o
/Users/togo/spglib/fortran/spglib_f08.F90:776:48:

  776 |         magspgtype%og_number = trim(magstype_c%og_number)
      |                                                1
Error: Symbol ‘magstype_c’ at (1) has no IMPLICIT type
```

Reply:
Typo on my side:
```fortran
magspgtype%og_number = trim(magspgtype_c%og_number)
```

Reply:
Oh yes. I was not careful. But what I wanted to show was
```
[ 50%] Building Fortran object fortran/CMakeFiles/Spglib_fortran.dir/spglib_f08.F90.o
/Users/togo/spglib/fortran/spglib_f08.F90:776:36:

  776 |         magspgtype%og_number = trim(magspgtype_c%og_number)
      |                                    1
Error: ‘string’ argument of ‘trim’ intrinsic at (1) must be a scalar
```